datasets/imppres/imppres.py

```diff
@@ -0,0 +1,318 @@
+
+    VERSION = datasets.Version("1.1.0")
+
+    # This is an example of a dataset with multiple configurations.
+    # If you don't want/need to define several sub-sets in your dataset,
+    # just remove the BUILDER_CONFIG_CLASS and the BUILDER_CONFIGS attributes.
+
+    # If you need to make complex sub-parts in the datasets with configurable options
+    # You can create your own builder configuration class to store attribute, inheriting from datasets.BuilderConfig
+    # BUILDER_CONFIG_CLASS = MyBuilderConfig
+
+    # You will be able to load one or the other configurations in the following list with
+    # data = datasets.load_dataset('my_dataset', 'first_domain')
+    # data = datasets.load_dataset('my_dataset', 'second_domain')
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(
+            name="presupposition",
+            version=VERSION,
+            description="Presuppositions are facts that the speaker takes for granted when uttering a sentence.",
+        ),
+        datasets.BuilderConfig(
+            name="implicature",
+            version=VERSION,
+            description="Scalar implicatures are inferences which can be drawn when one member of a memorized lexical scale is uttered.",
+        ),
+    ]
+
+    # DEFAULT_CONFIG_NAME = "first_domain"  # It's not mandatory to have a default configuration. Just use one if it make sense.
+
+    def _info(self):
+        if (
+            "presupposition" in self.config.name
+        ):  # This is the name of the configuration selected in BUILDER_CONFIGS above
+            features = datasets.Features(
+                {
+                    "premise": datasets.Value("string"),
+                    "hypothesis": datasets.Value("string"),
+                    "trigger": datasets.Value("string"),
+                    "trigger1": datasets.Value("string"),
+                    "trigger2": datasets.Value("string"),
+                    "presupposition": datasets.Value("string"),
+                    "gold_label": datasets.Value("string"),
+                    "UID": datasets.Value("string"),
+                    "pairID": datasets.Value("string"),
+                    "paradigmID": datasets.Value("int16")
+                    # These are the features of your dataset like images, labels ...
+                }
+            )
+        else:  # This is an example to show how to have different features for "first_domain" and "second_domain"
+            features = datasets.Features(
+                {
+                    "premise": datasets.Value("string"),
+                    "hypothesis": datasets.Value("string"),
+                    "gold_label_log": datasets.Value("string"),
+                    "gold_label_prag": datasets.Value("string"),
```

Comment:
same here

Not that you'll need to regenerate the dataset_infos.json file after these changes

Reply:
Done!